TrailBlaze/Networking/Interceptor.m

```diff
@@ -0,0 +1,125 @@
+    
+    MKPlacemark *startRunnerPlacemark = [[MKPlacemark alloc] initWithCoordinate:CLLocationCoordinate2DMake(runnerLocation.coordinate.latitude, runnerLocation.coordinate.longitude)];
+    
+    MKMapItem *startInterceptorItem = [[MKMapItem alloc] initWithPlacemark:startInterceptorPlacemark];
+    
+    MKMapItem *startRunnerItem = [[MKMapItem alloc] initWithPlacemark:startRunnerPlacemark];
+
+    for (int i=0; i < prunedPoints.count; i++) {
+        NSLog(@"starting operation %@..", @(i));
+        
+        dispatch_group_enter(group);
+        MKPlacemark *destinationPlacemark = [[MKPlacemark alloc] initWithCoordinate:CLLocationCoordinate2DMake([[prunedPoints objectAtIndex:i][0] doubleValue], [[prunedPoints objectAtIndex:i][1] doubleValue])];
+
+        MKMapItem *destinationItem =  [[MKMapItem alloc] initWithPlacemark:destinationPlacemark];
+
+        MKDirectionsRequest *pathInterceptorRequest = [[MKDirectionsRequest alloc] init];
+        [pathInterceptorRequest setSource:startInterceptorItem];
+        [pathInterceptorRequest setDestination:destinationItem];
+        [pathInterceptorRequest setTransportType:MKDirectionsTransportTypeWalking];
+        [pathInterceptorRequest setRequestsAlternateRoutes:NO];
+        
+        MKDirectionsRequest *pathRunnerRequest = [[MKDirectionsRequest alloc] init];
+        [pathRunnerRequest setSource:startRunnerItem];
+        [pathRunnerRequest setDestination:destinationItem];
+        [pathRunnerRequest setTransportType:MKDirectionsTransportTypeWalking];
+        [pathRunnerRequest setRequestsAlternateRoutes:NO];
+        
+        MKDirections *pathInterceptor = [[MKDirections alloc] initWithRequest:pathInterceptorRequest];
+        MKDirections *pathRunner = [[MKDirections alloc] initWithRequest:pathRunnerRequest];
+        [pathInterceptor calculateETAWithCompletionHandler:^(MKETAResponse * _Nullable interceptorResponse, NSError * _Nullable error1) {
+            if (interceptorResponse) {
+                [pathRunner calculateETAWithCompletionHandler:^(MKETAResponse * _Nullable runnerResponse, NSError * _Nullable error2) {
+                    if (runnerResponse) {
+                        [etasDifferences addObject:[NSNumber numberWithDouble:fabs(interceptorResponse.expectedTravelTime - runnerResponse.expectedTravelTime)]];
+                        [etaPointPairs setValue:runnerResponse.destination forKey:[NSString stringWithFormat:@"%.f", fabs(interceptorResponse.expectedTravelTime - runnerResponse.expectedTravelTime)]];
+                    } else {
+                        NSLog(@"%@", error2.localizedDescription);
+                    }
+                    dispatch_async(serialQueue, ^{
+                        dispatch_group_leave(group);
+                    });
+                }];
+            } else {
+                NSLog(@"%@", error1.localizedDescription);
```

Comment:
you'll also need to `dispatch_group_leave` in this error case here, otherwise you'll have `enter`s which are left open